### ISSUE TITLE
What does the confident map mean?

### ISSUE THREAD

**Issue by John1231983** [hfslyc/AdvSemiSeg#4]
It is not a bug. I just want to ask this question to make more clearly understand.
Your discriminator network produces a spatial map by using fully convolution layer. 
1. Does the spatial map same as the confident map? Or confident map generates by unlabeled data? 
2. Does the value of the spatial map (from D network) in range 0 to 1 or just {0,1}?

**Comment by hfslyc** [comment]
The output of D is a probability map with values ranges from 0-1. When the input is unlabeled data, we use the D output map as the indicator for semi-supervised learning.

**Comment by John1231983** [comment]
Thanks for your reply. So, what does confident map mean( in question 1)? Is it just ouput of D?

**Comment by chuanruihu** [comment]
Hi,  when I train the model, I find that the loss_seg is always about 1.5 shock. The loss_adv_p is  up to a few dozen(80).  The loss_D is about 0.1.  I just train the model without pretrained model. Your code train.py line 194 to line 207  I just log off to train the new model. Can you tell me how to train a new model.  Thank you !

**Comment by John1231983** [comment]
Me too. I also interested how to train the network from scratch.

**Comment by rhp228228** [comment]
Me too.   The code  may be has some wrong.  I am not clearly.  I also want to know how to train the network from scratch.

**Comment by hfslyc** [comment]
Hi all, I'm investigating the problem. Something might go wrong when I was cleaning the code.

**Comment by chuanruihu** [comment]
Thank you!  Your work is really interesting . I look forward you can check the code quickly, and can write the README detailedly. thank you.

**Comment by hfslyc** [comment]
@chuanruihu how do you train the model without pretrained model? The training process should always start from a imagenet pretrained model. Otherwise, it will not converge.

**Comment by chuanruihu** [comment]
@hfslyc   Hi,  I train the model without pretrained model, is just log off(注释) line194 to line207 in train.py. After you reply ,I konw your code is must start from a imagenet pretained model. Thank you for your reply. 

Second! when I just trained the model, I find that loss_semi is always in {0.001, 0.002, 0.003...}. I don't know   
 why the loss_semi is always this number.  and the loss_semi is what ?

**Comment by hfslyc** [comment]
@chuanruihu please refer to our paper for detailed description for semi-supervised loss

**Comment by hfslyc** [comment]
Hi all, if there is no further issue. I'm closing this one. Again, thank you for interested in our work.

**Comment by John1231983** [comment]
@hfslyc : Sorry for reply late. Could you guide me how could we obtain the figure of confidence map in your figure? Just feed the prediction to the trained weighted of D network? Do we need to use argmax()
